common/src/java/org/apache/hive/http/HttpServer.java

```diff
@@ -135,9 +136,10 @@ public class HttpServer {
 
 
   private final String name;
-  private String appDir;
-  private WebAppContext webAppContext;
+  private List<WebAppContext> webAppContexts;
   private Server webServer;
+  private QueuedThreadPool threadPool;
```

Comment:
why do you need this? can't you just use `builder#maxThreads`?

Reply:
`builder#maxThreads` is not always defined:

```
    threadPool = new QueuedThreadPool();
    if (b.maxThreads > 0) {
      threadPool.setMaxThreads(b.maxThreads);
    }
```